tests/check_framework/tests.py

```diff
@@ -305,3 +305,81 @@ class ModelWithDescriptorCalledCheck(models.Model):
 class ChecksRunDuringTests(SimpleTestCase):
     def test_registered_check_did_run(self):
         self.assertTrue(my_check.did_run)
+
+
+class CheckDuplicateDBTableTests(SimpleTestCase):
+
+    @isolate_apps('check_framework', kwarg_name='apps')
+    @override_system_checks([checks.model_checks.check_all_models])
+    def test_db_table_app_collissions_check_within_same_app(self, apps):
+
+        class ModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                db_table = 'test_table'
+
+        class AnotherModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                db_table = 'test_table'
+
+        errors = checks.run_checks(app_configs=apps.get_app_configs())
+        expected = [
+            Error(
+                "db_table 'test_table' was declared in multiple models: "
+                "'check_framework.ModelWithDBTableDefined, "
+                "check_framework.AnotherModelWithDBTableDefined'",
+                obj='test_table',
+                id='models.E028'
+            ),
+        ]
+        self.assertEqual(errors, expected)
+
+    @modify_settings(INSTALLED_APPS={'append': 'admin_inlines'})
+    @isolate_apps('check_framework', 'admin_inlines', kwarg_name='apps')
+    @override_system_checks([checks.model_checks.check_all_models])
+    def test_db_table_app_collissions_check_across_apps(self, apps):
+
+        class ModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                app_label = 'check_framework'
+                db_table = 'test_table'
+
+        class AnotherModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                app_label = 'admin_inlines'
+                db_table = 'test_table'
+
+        errors = checks.run_checks(app_configs=apps.get_app_configs())
+        expected = [
+            Error(
+                "db_table 'test_table' was declared in multiple models: "
+                "'check_framework.ModelWithDBTableDefined, "
+                "admin_inlines.AnotherModelWithDBTableDefined'",
+                obj='test_table',
+                id='models.E028'
+            ),
+        ]
+        self.assertEqual(errors, expected)
+
+    @isolate_apps('check_framework', kwarg_name='apps')
+    @override_system_checks([checks.model_checks.check_all_models])
+    def test_db_table_app_no_collissions_for_unmanaged_models(self, apps):
+
+        class ModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                db_table = 'test_table'
+                managed = False
+
+        class AnotherModelWithDBTableDefined(models.Model):
+
+            class Meta:
+                db_table = 'test_table'
+                managed = False
```

Comment:
You want to test with a managed and one unmanaged model, not two unmanaged models.